### DISCUSSION TITLE
Assertions for `StatusCode` on `HttpResponseMessage`

### DISCUSSION THREAD

**Discussion by mu88** [fluentassertions/fluentassertions#1698] [Ideas]
I'm using Fluent Assertions for the integration / E2E tests of my web API as well. So I have a couple of assertions which look like `response.StatusCode.Should().Be(HttpStatusCode.Ok)`.

What do you think about some assertions for `StatusCode` on `HttpResponseMessage` like:
- `BeSuccessStatusCode()`
- `BeFailedStatusCode()`
- `BeStatusCode(HttpStatusCode expectedStatusCode)`

If you like the idea, I'd love to provide a PR.

**Comment by dennisdoomen** [comment]
It depends. If it requires FA to take additional dependencies. But maybe it would be a better fit for https://github.com/fluentassertions/fluentassertions.aspnetcore.mvc cc @kevinkuszyk

**Reply by mu88** [reply]
As far as I see, `HttpResponseMessage` is in `System.Net.Http` and `HttpStatusCode` is in `System.Net`. Correct me if I'm wrong, but both should be already there, right?

But as you prefer, @dennisdoomen

**Reply by mu88** [reply]
Hey @dennisdoomen and @kevinkuszyk ,
What are your thoughts about this? If there is no interest in these assertions I'd stick with my project-specific ones and close the question to not pollute this repo with orphaned questions 🙂

**Reply by dennisdoomen** [reply]
As I said earlier, I think it should fit well with what @kevinkuszyk has been doing. But I'll let him respond first.

> close the question to not pollute this repo with orphaned questions

It's a discussion, so don't worry about that. And you can't close them anyway ;-)

**Reply by kevinkuszyk** [reply]
Apologies the the delay in replying.

I'm a little out of the loop in all things .net these days.  It does sound like a good fit for the MVC assertions, and I _think_ we have assertions for these already.

/cc @faddiv who is looking after the MVC extensions now.

**Reply by faddiv** [reply]
fluentassertions.aspnetcore.mvc deals specifically with the asp.net core classes like action result so it depends on these libs. This seems more like a separate thing.

**Reply by mu88** [reply]
Meaning that you would see the proposed assertions for `StatusCode` not within `FluentAssertions.AspNetCore.MVC`?

**Reply by faddiv** [reply]
I see two problems with it:
HttpResponseMessage is part of the System.Net.Http, which is a core assembly. FluentAssertions.AspNetCore.MVC depends on Microsoft.AspNetCore.Mvc.Core and therefore it is an unnecessary dependency in this case.
Also, I think this is a different kind of testing than which the FluentAssertions.AspNetCore.MVC aims for.
What is your opinion @dennisdoomen? Do you think this is still ok?

**Reply by dennisdoomen** [reply]
Yeah, you're right. So I think it's fine to add them to our core library after all.

**Reply by mu88** [reply]
Okay guys, thank you for the positive discussion :) so I will provide a PR with my existing assertions.
